widgetssdk/src/main/java/com/glia/widgets/entrywidget/EntryWidget.kt

```diff
@@ -37,11 +37,11 @@ internal class EntryWidgetImpl(
     private val activityLauncher: ActivityLauncher,
     private val themeManager: UnifiedThemeManager,
     private val entryWidgetHideController: EntryWidgetHideController,
-    private val secureConversationsRepository: SecureConversationsRepository
+    private val hasPendingSecureConversationsWithTimeoutUseCase: HasPendingSecureConversationsWithTimeoutUseCase
 ) : EntryWidget {
 
     override fun show(activity: Activity) {
-        secureConversationsRepository.getHasPendingSecureConversationsWithTimeout().unSafeSubscribe {
+        hasPendingSecureConversationsWithTimeoutUseCase().unSafeSubscribe {
```

Comment:
Does it mean the visitor can see a delay when they click some button and see the result?

Reply:
Since this property should be present right after the initialization in the `visitor_state,` we will not have a delay. But if it is somehow missing, the default `false` value will be returned after 1 second of waiting.

Reply:
Why can't we display the Entry Widget loading state and, after everything is prepared, go to the SC or display the available media types?

Reply:
I think that will not be user-friendly. So, the visitors will see the entry widget with the loading state, and then the entry widget will disappear, and the chat screen will appear automatically. That might be not very clear.
Besides that, this will happen in very, very corner cases when the system is healthy, but this attribute is somehow missing from the `visitor_state` or, for some reason, the socket disconnected, so it is safe to just open the entry widget.

Reply:
/Turning on the QA mode

It doesn't matter to me how often it happens, as long as it's possible. Displaying the loading screen is way more user-friendly than nothing to display. Moreover, we do not even provide some callback interfaces for integrators.

/Turning off the QA mode

Nice solution! :)

Reply:
> It doesn't matter to me how often it happens, as long as it's possible. Displaying the loading screen is way more user-friendly than nothing to display

I'm sorry Andrii's QA mode 🙂, but I do not agree with this statement.
The use case is to open the chat directly, instead of the entry widget, if there is pending SC. This should be very smooth for the users. I think a 1-second timeout is not enough to make visible delays for users, but if you insist, we can decrease that time or return the default value immediately if there is no value.


> Moreover, we do not even provide some callback interfaces for integrators

But in which cases integrators will need to know about pending SC?


> Nice solution! :)

Thanks 💟